internal/metrics/metrics-emitter.go

```diff
@@ -136,8 +136,13 @@ func (emitter *MetricEmitter) tryToSendOut() shutdown.Condition {
 	if err == nil {
 		return nil
 	}
-
-	return shutdown.Metric(fmt.Sprintf("failed to send metrics: %v", err))
+	message := fmt.Sprintf("failed to send metrics: %v", err)
+	if failFast {
+		return shutdown.Metric(message)
+	} else {
+		log.Println("Could not send metrics, but failFast disabled:", message)
```

Comment:
I would consider simply 

```suggestion
		log.Println(message)
```